pkg/generate/code/set_resource_test.go

```diff
@@ -3273,6 +3273,336 @@ func TestSetResource_EC2_DHCPOptions_NestedSetConfig(t *testing.T) {
+			kof10.DBUser = resp.RedshiftDataParameters.DbUser
+		}
+		if resp.RedshiftDataParameters.SecretManagerArn != nil {
+			kof10.SecretManagerARN = resp.RedshiftDataParameters.SecretManagerArn
+		}
+		if resp.RedshiftDataParameters.Sql != nil {
+			kof10.Sql = resp.RedshiftDataParameters.Sql
+		}
+		if resp.RedshiftDataParameters.StatementName != nil {
+			kof10.StatementName = resp.RedshiftDataParameters.StatementName
+		}
+		if resp.RedshiftDataParameters.WithEvent != nil {
+			kof10.WithEvent = resp.RedshiftDataParameters.WithEvent
+		}
+		ko.RedshiftDataParameters = kof10
+	}
+	if resp.RetryPolicy != nil {
+		kof11 := &svcapitypes.RetryPolicy{}
+		if resp.RetryPolicy.MaximumEventAgeInSeconds != nil {
+			kof11.MaximumEventAgeInSeconds = resp.RetryPolicy.MaximumEventAgeInSeconds
+		}
+		if resp.RetryPolicy.MaximumRetryAttempts != nil {
+			kof11.MaximumRetryAttempts = resp.RetryPolicy.MaximumRetryAttempts
+		}
+		ko.RetryPolicy = kof11
+	}
+	if resp.RoleArn != nil {
+		ko.RoleARN = resp.RoleArn
+	}
+	if resp.RunCommandParameters != nil {
+		kof13 := &svcapitypes.RunCommandParameters{}
+		if resp.RunCommandParameters.RunCommandTargets != nil {
+			kof13f0 := []*svcapitypes.RunCommandTarget{}
+			for _, kof13f0iter := range resp.RunCommandParameters.RunCommandTargets {
+				kof13f0elem := &svcapitypes.RunCommandTarget{}
+				if kof13f0iter.Key != nil {
+					kof13f0elem.Key = kof13f0iter.Key
+				}
+				if kof13f0iter.Values != nil {
+					kof13f0elemf1 := []*string{}
+					for _, kof13f0elemf1iter := range kof13f0iter.Values {
+						var kof13f0elemf1elem string
+						kof13f0elemf1elem = *kof13f0elemf1iter
+						kof13f0elemf1 = append(kof13f0elemf1, &kof13f0elemf1elem)
+					}
+					kof13f0elem.Values = kof13f0elemf1
+				}
+				kof13f0 = append(kof13f0, kof13f0elem)
+			}
+			kof13.RunCommandTargets = kof13f0
+		}
+		ko.RunCommandParameters = kof13
+	}
+	if resp.SageMakerPipelineParameters != nil {
+		kof14 := &svcapitypes.SageMakerPipelineParameters{}
+		if resp.SageMakerPipelineParameters.PipelineParameterList != nil {
+			kof14f0 := []*svcapitypes.SageMakerPipelineParameter{}
+			for _, kof14f0iter := range resp.SageMakerPipelineParameters.PipelineParameterList {
+				kof14f0elem := &svcapitypes.SageMakerPipelineParameter{}
+				if kof14f0iter.Name != nil {
+					kof14f0elem.Name = kof14f0iter.Name
+				}
+				if kof14f0iter.Value != nil {
+					kof14f0elem.Value = kof14f0iter.Value
+				}
+				kof14f0 = append(kof14f0, kof14f0elem)
+			}
+			kof14.PipelineParameterList = kof14f0
+		}
+		ko.SageMakerPipelineParameters = kof14
+	}
+	if resp.SqsParameters != nil {
+		kof15 := &svcapitypes.SQSParameters{}
+		if resp.SqsParameters.MessageGroupId != nil {
+			kof15.MessageGroupID = resp.SqsParameters.MessageGroupId
+		}
+		ko.SQSParameters = kof15
+	}
+`
+	assert.Equal(
+		expected,
+		code.SetResourceForStruct(
+			crd.Config(), crd, "", "ko", koShape, nil, "resp", targetShape, "", model.OpTypeList, 1,
```

Comment:
Turns out that the [`targetField`](https://github.com/aws-controllers-k8s/code-generator/blob/c6651c200ba136e5c7f50ad8be751fae060a38e6/pkg/generate/code/set_resource.go#L1258-L1259) parameter isn't ever used by the `SetResourceForStruct` function :( Only the [`targetFieldPath`](https://github.com/aws-controllers-k8s/code-generator/blob/c6651c200ba136e5c7f50ad8be751fae060a38e6/pkg/generate/code/set_resource.go#L1270-L1271) parameter is ever used.

My request is that you:

1) remove the unused `targetField` parameter (since it is just confusing) from the `SetResourceForStruct` function
2) Pass `fX` for the *targetVarName* parameter (normally this would be something like "f12" if `SetResourceForStruct` is being called recursively from other functions in `pkg/generate/code/set_resource.go` *or* it would be the name of the field if `SetResourceForStruct` was called from a top-level template in `sdk_file_end` hooks...)